sktime/forecasting/base/tests/test_base_bugs.py

```diff
@@ -58,3 +60,21 @@ def test_heterogeneous_get_fitted_params():
 
     reconciler.fit(y_agg)
     reconciler.get_fitted_params()  # triggers an error pre-fix
+
+
+@pytest.mark.skipif(
+    not _check_estimator_deps(SARIMAX, severity="none"),
+    reason="skip test if required soft dependency not available",
+)
+def test_predict_residuals_conversion():
+    """Regression test for bugfix #4766, related to predict_residuals internal type."""
+    from sktime.datasets import load_longley
+    from sktime.forecasting.model_selection import temporal_train_test_split
+
+    y, X = load_longley()
+    y_train, y_test, X_train, X_test = temporal_train_test_split(y, X)
+    pipe = Differencer() * SARIMAX()
```

Comment:
Not Blocking, but should we replace SARIMAX with NaiveForecaster? With that change, this test would also be executed if the soft dependency is not available.

Reply:
Hm, perhaps - although `NaiveForecaster` ignores `X`. The issue could also be specific to the inner types (I didn't check), so I simply went with a known failure.

Did you ccheck whether `NaiveForecaster` also causes the issue (pre-PR)?

Reply:
I tested it a minute ago and 

```python
   from sktime.datasets import load_longley
   from sktime.forecasting.model_selection import temporal_train_test_split
   
   y, X = load_longley()
   y_train, y_test, X_train, X_test = temporal_train_test_split(y, X)
   pipe = Differencer() * NaiveForecaster()
   pipe.fit(y=y_train, X=X_train, fh=[1, 2, 3, 4])
   result = pipe.predict_residuals()
   
```

results in the same error.

Reply:
ok, replaced with `NaiveForecaster` now.